### DISCUSSION TITLE
Add row-based selection to QTable

### DISCUSSION THREAD

**Discussion by ddenev** [quasarframework/quasar#13359] [Ideas / Proposals]
Would it be possible to add a built-in way to make the rows selected by clicking on them, rather than clicking only on the selection checkboxes?

My use-case:
Currently I implement row-based selection by utilizing the `@row-click` event and manipulating the `selected` prop and this does work well for single selection. Now I want to add Ctrl and Shift modifiers to implement multi-selection (as in this example - https://quasar.dev/vue-components/table#example--custom-multiple-selection) and I need to use the `@selection` event.
The problem is that this event is called only when I'm selecting by clicking on the checkboxes.

Alternatively, I was also thinking whether adding a method, e.g. `select`, to QTable, which I would call, and which in turn would fire the `@selection` event - would solve my problem.

**Comment by ddenev** [comment]
For the alternative idea above, I just checked that QTable.js actually has a method `updateSelection` which would serve the purpose but it's not exposed on the public interface of QTable.

**Reply by adametcvadim** [reply]
You wrote: "Currently I implement row-based selection by utilizing the `@row-click` event and manipulating the `selected` prop and this does work well for single selection. "
How do you do it?
Do I understand correctly that I need to pass to the `selected` row object under the number `@row-click`?

**Comment by ddenev** [comment]
bump

**Comment by ddenev** [comment]
bump

**Reply by metalsadman** [reply]
have you tried taking a stab on it locally? a pr maybe.
